bundles/org.eclipse.swt/Eclipse SWT Tests/win32/org/eclipse/swt/widgets/WidgetWin32Tests.java

```diff
@@ -28,8 +36,280 @@ public void testWidgetZoomShouldChangeOnZoomLevelChange() {
+		assertEquals("Height should be half in points after zooming to 200", p1.y, p2.y * 2);
+	}
+
+	@Test
+	public void testImagePixelsWithDoubleZoomLevel() {
+		int zoom = DPIUtil.getDeviceZoom();
+		int scaledZoom = zoom * 2;
+
+		InputStream inputStream = WidgetWin32Tests.class.getResourceAsStream("folder.png");
+		Image image = new Image(display, inputStream);
+
+		Point buttonImageSizeBeforeEvent = getImageDimension(image, zoom);
+		Point buttonImageSizeAfterEvent = getImageDimension(image, scaledZoom);
+
+		assertEquals("Width of a button image should be doubled after zooming to 200",
+				buttonImageSizeBeforeEvent.x * 2, buttonImageSizeAfterEvent.x);
+		assertEquals("Height of a button image should be doubled after zooming to 200",
+				buttonImageSizeBeforeEvent.y * 2, buttonImageSizeAfterEvent.y);
+	}
+
+	public Point getImageDimension(Image image, Integer zoomLevel) {
+		BITMAP bm = new BITMAP();
+		OS.GetObject(Image.win32_getHandle(image, zoomLevel), BITMAP.sizeof, bm);
+		int imgWidth = bm.bmWidth;
+		int imgHeight = bm.bmHeight;
+		return new Point(imgWidth, imgHeight);
+	}
+
+	@Test
+	public void testButtonFontAfterZooming() {
+		int zoom = DPIUtil.getDeviceZoom();
+		int scaledZoom = zoom * 2;
+
+		shell = new Shell(display);
+		shell.setBounds(0, 0, 100, 160);
+		shell.setLayout(new FillLayout());
+		shell.pack();
+
+		Button button = new Button(shell, SWT.PUSH);
+		button.setText("Button");
+		button.setBounds(0, 0, 100, 200);
+		Font font = new Font(display, "Arial", 12, SWT.BOLD);
+		button.setFont(font);
+
+		int heightBeforeZoom = button.getFont().getFontData()[0].data.lfHeight;
+		changeDPIZoom(scaledZoom);
+		int heightAfterZoom = button.getFont().getFontData()[0].data.lfHeight;
+
+		assertEquals("Height of a font of the button should be doubled after zooming to 200",
+				heightBeforeZoom * 2, heightAfterZoom);
+	}
+
+	@Test
+	public void testCoolItemAfterZooming() {
+		int zoom = DPIUtil.getDeviceZoom();
+		int scaledZoom = zoom * 2;
+
+		shell = new Shell(display);
+		shell.setBounds(0, 0, 100, 160);
+		shell.setLayout(new FillLayout());
+		shell.pack();
+
+		CoolBar coolBar = new CoolBar(shell, SWT.NONE);
+		CoolItem item1 = new CoolItem(coolBar, SWT.NONE);
+		Label label1 = new Label(coolBar, SWT.NONE);
+		label1.setText("Label 1");
+		item1.setControl(label1);
+		item1.setPreferredSize(label1.computeSize(SWT.DEFAULT, SWT.DEFAULT));
+		item1.setSize(item1.getPreferredSize());
+
+		var preferredControlSize = item1.getPreferredSizeInPixels();
+		int xBeforeZoom = preferredControlSize.x;
+		int yBeforeZoom = preferredControlSize.y;
+		changeDPIZoom(scaledZoom);
+		var preferredControlSize2 = item1.getPreferredSizeInPixels();
+		int xAfterZoom = preferredControlSize2.x;
+		int yAfterZoom = preferredControlSize2.y;
+
+		assertTrue("Height of a Item should be greater after zooming to 200", xBeforeZoom < xAfterZoom);
+		assertTrue("Width of a Item should be greater after zooming to 200", yBeforeZoom < yAfterZoom);
```

Comment:
Why not twice as big like in other tests?

Reply:
Because for CoolItem label, the OS.DrawText call don't return exactly 2x for height. Since this value is coming from OS, I don't see any other option rather than just checking if the value is larger than before. I think we have discussed this before.